test/e2e/common_suite_test.go

```diff
@@ -36,104 +31,6 @@ import (
 const WAIT_POD_RUNNING_TIMEOUT = time.Second * 900
 const WAIT_JOB_RUNNING_TIMEOUT = time.Second * 600
 
-// testCommand is a list of commands to execute inside the pod container,
-// each with a function to test if the command outputs the value the test
-// expects it to on the stdout stream
-type testCommand struct {
-	command             []string
-	testCommandStdoutFn func(stdout bytes.Buffer) bool
-	containerName       string
-}
-type instanceValidatorFunctions struct {
-	testSuccessfn func(instance string) bool
-	testFailurefn func(error error) bool
-}
-
-type testCase struct {
-	testing              *testing.T
-	testName             string
-	assert               CloudAssert
-	assessMessage        string
-	pod                  *v1.Pod
-	configMap            *v1.ConfigMap
-	secret               *v1.Secret
-	pvc                  *v1.PersistentVolumeClaim
-	job                  *batchv1.Job
-	testCommands         []testCommand
-	expectedPodLogString string
-	podState             v1.PodPhase
-	imagePullTimer       bool
-	isAuth               bool
-	AuthImageStatus      string
-	deletionWithin       *time.Duration
-	testInstanceTypes    instanceValidatorFunctions
-}
-
-func (tc *testCase) withConfigMap(configMap *v1.ConfigMap) *testCase {
-	tc.configMap = configMap
-	return tc
-}
-
-func (tc *testCase) withSecret(secret *v1.Secret) *testCase {
-	tc.secret = secret
-	return tc
-}
-
-func (tc *testCase) withPVC(pvc *v1.PersistentVolumeClaim) *testCase {
-	tc.pvc = pvc
-	return tc
-}
-
-func (tc *testCase) withJob(job *batchv1.Job) *testCase {
-	tc.job = job
-	return tc
-}
-
-func (tc *testCase) withPod(pod *v1.Pod) *testCase {
-	tc.pod = pod
-	return tc
-}
-
-func (tc *testCase) withDeleteAssertion(duration *time.Duration) *testCase {
-	tc.deletionWithin = duration
-	return tc
-}
-
-func (tc *testCase) withTestCommands(testCommands []testCommand) *testCase {
-	tc.testCommands = testCommands
-	return tc
-}
-
-func (tc *testCase) withInstanceTypes(testInstanceTypes instanceValidatorFunctions) *testCase {
-	tc.testInstanceTypes = testInstanceTypes
-	return tc
-}
-
-func (tc *testCase) withExpectedPodLogString(expectedPodLogString string) *testCase {
-	tc.expectedPodLogString = expectedPodLogString
-	return tc
-}
-
-func (tc *testCase) withCustomPodState(customPodState v1.PodPhase) *testCase {
-	tc.podState = customPodState
-	return tc
-}
-
-func (tc *testCase) withPodWatcher() *testCase {
-	tc.imagePullTimer = true
-	return tc
-}
-
-func (tc *testCase) withAuthenticatedImage() *testCase {
-	tc.isAuth = true
-	return tc
-}
-
-func (tc *testCase) withAuthImageStatus(status string) *testCase {
-	tc.AuthImageStatus = status
-	return tc
-}
-
 func (tc *testCase) run() {
```

Comment:
Should testCase run move out somewhere as well?

Reply:
Hi @stevenhorsman ,
I just moved run() method to different file, but there  [testEnv](https://github.com/confidential-containers/cloud-api-adaptor/pull/1420/commits/f2680539058854d92ba2954f3997ba2ce8a68a8b#diff-108219a173223ecaadfe6371b6295385a0fbe7278314ad4f7e768b2c3e9536a9R399) become `undecalred`, once I changed the file name with `_test` suffix, it got resolved.

Reply:
I think that's because go unit test files `_test.go` are compiled separately and will try to be executed as tests. I'm not a golang expert though, so someone else might be able to give better advice